lambdas/db-migration/src/migrations/20200901060126_create_granules.ts

```diff
@@ -0,0 +1,31 @@
+import * as Knex from 'knex';
+
+export const up = async (knex: Knex): Promise<void> => knex.schema
+  .createTable('granules', (table) => {
+    table.bigIncrements('cumulusId').comment('Internal Cumulus ID for a granule').primary();
+    table.integer('collectionCumulusId').references('cumulusId').inTable('collections').notNullable();
+    table.timestamps(false, true);
+    table.boolean('published').comment('Flag that shows if the granule has been published in CMR');
+    table.enum('status', ['running', 'completed', 'failed']).comment('Ingest status of the granule');
+    table.float('duration').comment('Ingest duration');
+    table.float('timeToArchive').comment('Number of seconds granule took to archive');
+    table.float('timeToProcess').comment('Number seconds granule took to complete "processing"');
+    table.integer('productVolume');
+    table.jsonb('error').comment('JSON error object');
+    table.text('cmrLink').comment('Link to granule in the CMR API');
+    table.text('execution').comment('Step Function Execution link');
+    table.text('granuleId').comment('Granule ID');
+    table.text('pdrName').comment('PDR associated with the granule');
+    table.text('provider').comment('Provider granule is associated with');
+    table.timestamp('beginningDateTime').comment('Date granule started');
+    table.timestamp('endingDateTime').comment('Date granule completed');
+    table.timestamp('lastUpdateDateTime').comment('Timestap for last update');
+    table.timestamp('processingEndDateTime').comment('Date granule finished processing');
+    table.timestamp('procvessingStartDateTime').comment('Date granule started processing');
+    table.timestamp('productionDateTime').comment('Timestamp for granule production date/time');
+    table.timestamp('timestamp');
+    table.unique(['granuleId', 'collectionCumulusId']);
+  });
+
+export const down = async (knex: Knex): Promise<void> => knex.schema
```

Comment:
What happens here if the table does not exist?

Reply:
It fails - should be we considering idempotent behavior for these initial table creation migrations?   It feels not great to have the database in an unexpected state for a rollback and have that rollback succeed, although we can explicitly write logic to handle that where needed as well.

Reply:
My vote is that a rollback that is trying to delete a table should succeed even if that table doesn't exist.

Reply:
I don't see an issue with it here.  I'm significantly more concerned about the wisdom of implementing that case for more complex schema migrations versus something like a snapshot restoration, but I can save that for a future arch topic.

Reply:
Knex to the rescue: http://knexjs.org/#Schema-dropTableIfExists

Reply:
@markdboyd  right -  it's not that the implementation is complex, it's more what we want to be doing moving forward.     No argument on the point on this PR though.

Reply:
8550c165b